src/elements/common/media-query/withMedia.js

```diff
@@ -0,0 +1,79 @@
+// @flow
+
+import * as React from 'react';
+import useMedia from './useMedia';
+import {
+    ANY_HOVER,
+    ANY_POINTER_COARSE,
+    ANY_POINTER_FINE,
+    HOVER,
+    POINTER_COARSE,
+    POINTER_FINE,
+    SIZE_SMALL,
+    SIZE_MEDIUM,
+    SIZE_LARGE,
+    SIZE_XLARGE,
+    POINTER_TYPE_NONE,
+    POINTER_TYPE_COARSE,
+    POINTER_TYPE_FINE,
+    HOVER_TYPE_NONE,
+    HOVER_TYPE_HOVER,
+} from './constants';
+
+type PropsShape = {
+    children: React.Node,
+};
+
+type PointerType = POINTER_TYPE_NONE | POINTER_TYPE_FINE | POINTER_TYPE_COARSE;
+
+type MediaShape = {
+    anyHover: 'none' | 'hover',
+    anyPointer: 'none' | 'fine' | 'coarse',
+    hover: 'none' | 'hover',
+    isExtraLarge: boolean,
+    isLarge: boolean,
+    isMedium: boolean,
+    isSmall: boolean,
```

Comment:
Can a page be more than one of these at a time? It seems like an enum for `size` would be better than a series of booleans.

Reply:
Switched to using enum strings for size